rrrspec-client/lib/rrrspec/client/slave_runner.rb

```diff
@@ -57,9 +56,12 @@ def work
           @timeout = TASKQUEUE_TASK_TIMEOUT
           trial = Trial.create(task, @slave)
 
-          @rspec_runner.reset
+          task_file_ext = File.extname(task.spec_file)
+          runner = RRRSpec::Registry.create_runner(task_file_ext)
```

Comment:
Renamed the RRRSpec::Registry.create_runner to RRRSpec::Registry.get_runner and there are no factories now. I believe the semantics are clearer now.